### ISSUE TITLE
Koala 2.1.3 has trojan? (Gen:Heur.Conjar.9)

### ISSUE THREAD

**Issue by pckltr** [oklai/koala#617]
C:\Program Files (x86)\Koala\Koala.exe is malware of type Gen:Heur.Conjar.9

**Comment by AleRob90** [comment]
Windows 10 defender was telling me that something is suspicious with installer.

**Comment by zaygraveyard** [comment]
Does [NW.js](https://dl.nwjs.io/v0.22.1/nwjs-v0.22.1-win-ia32.zip) have a trojan?

**Comment by timeceeper** [comment]
2.1.3 and 2.1.4 Koala.exe is deleted from virusscanner.

Category:	Malware
Type:	Other
Platform:	W32
Aliases:	Heuristic, Gen:Heur, Gen:Trojan.Heur, Memscan:trojan, Could be a mass-mailing worm, Could be infected with an unknown virus

**Comment by zaygraveyard** [comment]
@timeceeper Does [NW.js](https://dl.nwjs.io/v0.22.1/nwjs-v0.22.1-win-ia32.zip) have it?

**Comment by AleRob90** [comment]
@zaygraveyard only koala.exe was detected for me. Today i updated and the setup was canceled first time because of koala.exe. Second time it worked. Hope it helps

**Comment by zaygraveyard** [comment]
Can anyone please check [NW.js](https://dl.nwjs.io/v0.22.1/nwjs-v0.22.1-win-ia32.zip)
koala.exe is nwjs.exe with minor modifications so if nwjs is clean it means that the build system is the cause.
Thank you

**Comment by AleRob90** [comment]
@zaygraveyard I scanned with windows defender and nw.exe is clean. I can run it normally. Sorry for confusion.

**Comment by timeceeper** [comment]
Hello.. no Problem with nwjs-v0.22.1-win-ia32.

**Comment by zaygraveyard** [comment]
Thank you @AleRob90 and @timeceeper, I'm locking into it

**Comment by zaygraveyard** [comment]
Can you try to replace `koala.exe` with `nw.exe`:
1. Rename `koala.exe` to `koala.exe.bak`
2. Copy `nw.exe` to the Koala dir
3. Rename `nw.exe` to `koala.exe`

Then try to run it.
Thanks

**Comment by timeceeper** [comment]
Hello,
yes of couse. It is working. The app starts.

i have ignored the defender warning and installed it. The Antivirus have deleted the koala.exe.
i have copied the nw.exe, renamed it, and started it. The app starts.
i can see the v2.1.4 in the about info.

with regards

**Comment by zaygraveyard** [comment]
@timeceeper Thank you

To create `koala.exe` from `nw.exe`, I use [winresourcer](https://github.com/felicienfrancois/node-winresourcer) (which is a wrapper of [Anolis Resourcer](http://www.anol.is/resourcer/) running via [wine](http://www.winehq.org/) on my mac) to change the icon.

I'll try to use [rcedit](https://github.com/electron/node-rcedit) to do the same this and see if the result is also marked as trojan.

**Comment by lifenstein** [comment]
@zaygraveyard You may want to get a full system scan with multiple AVs; your system might be injecting these when the EXEs are being made.

**Comment by zaygraveyard** [comment]
@lifenstein I'm actually on a mac (and creating the packages on mac also) so I don't think it's machine is infecting the exe.
But I am going to try [rcedit](https://github.com/electron/node-rcedit), maybe the problem is coming from [winresourcer](https://github.com/felicienfrancois/node-winresourcer).

**Comment by lifenstein** [comment]
Are you saying Macs don't have viruses? You'd be surprised.

I've downloaded and extracted the Windows installers using 7zip, and Bitdefender found trojans in 2.14, 2.13 and 2.1.2. No infections were detected in 2.1.1.

**Comment by zaygraveyard** [comment]
@lifenstein Yes it is improbable, at this time, (but not impossible) that a mac is infected with a virus.
And even if it was I highly doubt it will infect a windows executable on my machine.

Can you scan v2.1.0?
`v2.1.0`, `v2.1.2`, and `v2.1.3` uses `nw.js` `v0.22.0`
`v2.1.1` uses `nw.js` `v0.21.6`
`v2.1.3` uses `nw.js` `v0.22.1`

**Comment by timeceeper** [comment]
Hello,
i have another (helpful?) information. 
Since i worked the last 2 days with the renamed nw.exe (nwjs-v0.22.1) the virusscanner (f-secure) delete the exe after a restart. It was very anoing, because the host-file is also be reseted. I wonder, because nw.exe download and copy/rename/start is not a problem.
The last version with no problem is the 2.1.1

**Comment by zaygraveyard** [comment]
I think it's a problem with NW.js and a false positive.

Relevant links:
* ["nw.exe" on EMSISoft's Is this file safe](http://www.isthisfilesafe.com/?filename=nw.exe)
* nwjs/nw.js#4023

**Comment by lifenstein** [comment]
No infections detected by BitDefender Free Edition in 2.1.0.

**Comment by lifenstein** [comment]
VirusTotal (multiple scanners) Scan results for Koala.exe:

2.1.0 - https://www.virustotal.com/en/file/d0ea32e3a228e9922653bb8bfd3a5c576a9f1e0248531cd53dc6621ca597a686/analysis/

2.1.1 - https://www.virustotal.com/en/file/48ac7eec5c1a74972aa61c142987ff6934ad2582e560303bb23e2b98ceca1c7e/analysis/

2.1.2 - https://www.virustotal.com/en/file/70b334ca9f192354db23987910e999e71079a5ef8b5d9f9176eb759789b6a95e/analysis/

2.1.3 -  (same file as above.)

2.1.4 - https://www.virustotal.com/en/file/1d4ffe5dba56e9c34719b8943fad4ecb932603e19709a14a15580c7799f98beb/analysis/

All except 2.1.0 are infected.

**Comment by zaygraveyard** [comment]
@lifenstein Thank you
That's interesting because v2.1.0 and v2.1.1 both where created with the `zip` option enabled, which means that all the resource files get bundled, zipped and concatenated to the original `nw.exe` (requiring a decompression on each lunch of the app resulting in a very slow lunch)
In the newer versions I disabled this option.

**Comment by zaygraveyard** [comment]
I tried to create the `Koala.exe` in a virtual Windows machine using Resource Hacker, the result got the same virus report as the original `nw.exe`
So the origin of this problem is probably `wine`
I'll try fix `wine`, but in the mean time I'll use my virtual machine to create the executable.

**Comment by zaygraveyard** [comment]
This should be fixed in v2.2.0

**Comment by lifenstein** [comment]
Yep, no issues. Thanks @zaygraveyard!

**Comment by AleRob90** [comment]
For me it's not working now at all. I downloaded from official the exe and now i cannot setup at all.
I have updated the defender an hour ago.
Until now i could pass the warning with suspicious, now it's removed instantly.

**Comment by timeceeper** [comment]
Sorry, it is not working on my PC, too. Same issu as before. Can not execute tne installer after downloading.
Windows Defender is blocking it.

**Comment by lifenstein** [comment]
https://www.virustotal.com/en/url/c2bf3c0f7e2afa941b6493adfbbe237222770a0052ed7ffde984bfc74632e54d/analysis/ says that 2.2.0 is clean. @AleRob90 , @timeceeper - your AVs' heuristics scanning might be detecting similar filename or patterns to guess that the new one also has it?

**Comment by lifenstein** [comment]
Just downloaded the file and uploaded to virustotal instead of giving the URL as in the scan result above, and 1 of 61 scanners found an issue.
https://www.virustotal.com/en/file/8c63cce1c0b2596d1bac1404d05e990528ade1263b3ee831a153da076af27a74/analysis/

**Comment by zaygraveyard** [comment]
@timeceeper does Windows Defender say why it's blocking it?

**Comment by timeceeper** [comment]
Hello,  no there is no info. (sorry is german OS)
![snap 2017-05-15 at 16 27 09](https://cloud.githubusercontent.com/assets/1217812/26062551/6fefd284-398b-11e7-88c2-c0aa09e700c3.png)

**Comment by zaygraveyard** [comment]
@timeceeper can you check the [History page of Windows Defender](https://www.microsoft.com/en-us/safety/pc-security/windows-defender.aspx)?

**Comment by zaygraveyard** [comment]
@timeceeper and if you don't find anything in the History page, maybe there's something in the [event logs](https://docs.microsoft.com/en-us/windows/threat-protection/windows-defender-antivirus/troubleshoot-windows-defender-antivirus)
Thank you

**Comment by zaygraveyard** [comment]
@timeceeper it appears that the problem is not that Koala is infected but that its not signed, see [this](https://www.codeproject.com/Questions/555248/Willplussigningplusanplusexecutablepluspreventplus).

**Comment by AleRob90** [comment]
@zaygraveyard for me nothing into the history page of Windows Defender at all and the event logs are irrelevant, updates only.

**Comment by timeceeper** [comment]
@zaygraveyard but are the old executables are signed? There are no entrys in the log.

**Comment by zaygraveyard** [comment]
@timeceeper before v2.1.0 I don't know if they were signed, but starting with v2.1.0 they are not signed.
